src/overlays/actors/ovl_Boss_Va/z_boss_va.c

```diff
@@ -173,157 +294,3728 @@ static ColliderQuadInit D_8095BFF4 = {
+                BODY->actor.world.rot.z += 0x4E2;
+                break;
+        }
+    }
+
+    Math_SmoothStepToS(&this->headRot.x, vaBody->vaBodySpinRate * -3, 1, 0x4B0, 0);
+    if (SkelAnime_Update(&this->skelAnime)) {
+        endFrame = Animation_GetLastFrame(&gBarinadeSupportDetachedAnim);
+        Animation_Change(&this->skelAnime, &gBarinadeSupportDetachedAnim, 1.0f, 0.0f, endFrame, ANIMMODE_LOOP_INTERP,
+                         0.0f);
+        this->actor.flags &= ~1;
+    }
+
+    if ((this->timer == 0) && (sCsState < DEATH_START)) {
+        this->timer = (s32)(Rand_ZeroOne() * 10.0f) + 10;
+        BossVa_BloodDroplets(globalCtx, &this->armTip, this->headRot.x, this->actor.shape.rot.y);
+    }
+
+    if (sCsState >= DEATH_START) {
+        Math_SmoothStepToF(&this->skelAnime.playSpeed, 0.0f, 0.3f, 0.25f, 0.125f);
+    }
+
+    switch (sCsState) {
+        case DEATH_SHELL_BURST:
+            sCameraEye = sCameraNextEye;
+            sCameraAt = sCameraNextAt;
+            Math_SmoothStepToF(&sCameraEye.x, sCameraNextAt.x, 1.0f, 10.0f, 0.0f);
+            Math_SmoothStepToF(&sCameraEye.z, sCameraNextAt.z, 1.0f, 10.0f, 0.0f);
+            sCameraEye.y += 20.0f;
+            sCsState++;
+
+        case DEATH_CORE_TUMORS:
+        case DEATH_CORE_DEAD:
+        case DEATH_CORE_BURST:
+            if (!this->burst) {
+                if ((globalCtx->gameplayFrames % 2) != 0) {
+                    BossVa_Tumor(globalCtx, this, 1, (s16)Rand_CenteredFloat(5.0f) + 6, 7.0f, 5.0f, TUMOR_ARM,
+                                 (this->timer2 >> 3) + 1, true);
+                }
+
+                this->timer2++;
+                if (this->timer2 >= 32) {
+                    this->burst++;
+                    this->isDead = true;
+                    Audio_PlayActorSound2(&this->actor, NA_SE_EN_BALINADE_BREAK2);
+                    if (this->actor.params == BOSSVA_SUPPORT_3) {
+                        sCsState++;
+                    }
+                }
+            } else {
+                this->timer2--;
+                if (this->timer2 == 0) {
+                    Actor_Kill(&this->actor);
+                }
+            }
+            break;
+    }
+
+    this->timer--;
+}
+
+void BossVa_SetupStump(BossVa* this, GlobalContext* globalCtx) {
+    Animation_Change(&this->skelAnime, &gBarinadeStumpAnim, 1.0f, 0.0f, Animation_GetLastFrame(&gBarinadeStumpAnim),
+                     ANIMMODE_ONCE, 0.0f);
+    this->actor.flags &= ~1;
+    BossVa_SetupAction(this, BossVa_Stump);
+}
+
+void BossVa_Stump(BossVa* this, GlobalContext* globalCtx) {
+    if (SkelAnime_Update(&this->skelAnime) && (Rand_ZeroOne() < 0.3f)) {
+        this->skelAnime.curFrame -= Rand_ZeroOne() * 3.0f;
+    }
+
+    if (sCsState >= DEATH_START) {
+        Actor_Kill(&this->actor);
+    }
+}
+
+void BossVa_SetupZapperIntro(BossVa* this, GlobalContext* globalCtx) {
+    f32 endFrame = Animation_GetLastFrame(&gBarinadeZapperIdleAnim);
```

Comment:
endFrame has since been renamed to lastFrame, so we can change all of the temps too (if there are more than just this one in the file)